### DISCUSSION TITLE
Why doesn't {RemoteIpAddress} placeholder work?

### DISCUSSION THREAD

**Discussion by SerkanKutlu** [ThreeMammals/Ocelot#1766] [Q&A] [answered]
## Expected Behavior / New Feature
I have a **ocelot.json** file like this;
```json
{
  "Routes": [
    {
      "DownstreamPathTemplate": "/api/Device/deneme1/{RemoteIpAddress}",
      "DownstreamScheme": "http",
      "DownstreamHostAndPorts": [
        {
          "Host": "newgateway",
          "Port": 80
        }
      ],
      "UpstreamHttpMethod": ["POST"],
      "UpstreamPathTemplate": "/Device"
    }
  ]
}
```

I need to get the id of client with using this. 

## Actual Behavior / Motivation for New Feature
The request url is : 
![image](https://user-images.githubusercontent.com/61331806/156914551-1bdce8af-cab2-4fd1-aad5-67c32505f8fb.png)

Placeholder is not working. What can I do? Thank you.

## Specifications

  - Version: 17.0.0
  - Platform: ASP.NET CORE 5.0
  - Subsystem:

**Comment by ks1990cn** [comment]
Not sure exactly what is HTTPMethod of downstream because you want to `GET` from downstream but mapped upstream is post call.
You can check again port and scheme of upstream and downstream.
here is doc https://ocelot.readthedocs.io/en/latest/features/routing.html

@SerkanKutlu Its been time you faced this issue, are you still facing same. If yes please let us know more about it.

Verified now on latest version that placeholder is working fine in every cases.  
@raman-m @ggnaegi

**Comment by raman-m** [comment, marked as answer]
@SerkanKutlu  Hi Serkan!
Because you are newbie to Ocelot! 😉 

You were confused by Route Placeholders feature and Headers Transform Placeholders one.
[Placeholders in paths](https://ocelot.readthedocs.io/en/latest/features/routing.html#placeholders) are parts of URL you want to extract from upstream URL and substitute to downstream URL.
`{RemoteIpAddress}` placeholder is the term of Headers Transformation feature, see more: [Headers Transformation | Placeholders](https://ocelot.readthedocs.io/en/latest/features/headerstransformation.html#placeholders).
Here is direct search link: [Search for {RemoteIpAddress}](https://ocelot.readthedocs.io/en/latest/search.html?q=%7BRemoteIpAddress%7D)
So, **RemoteIpAddress** value can be extracted from headers, so that is why this placeholder is a part of the Header Transformation.
**You cannot use `{RemoteIpAddress}` placeholder in paths!**
Or, you can use it, but for URL parts substitutions only, from upstream to downstream!
